pkg/test/e2e/cilium/cilium_test.go

```diff
@@ -0,0 +1,660 @@
+	"google.golang.org/grpc"
+
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/test/e2e/utils"
+	"k8c.io/kubermatic/v2/pkg/test/e2e/utils/apiclient/models"
+	yamlutil "k8c.io/kubermatic/v2/pkg/util/yaml"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	kjson "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/util/rand"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	userconfig      string
+	accessKeyID     string
+	secretAccessKey string
+)
+
+const (
+	seed            = "kubermatic"
+	projectName     = "cilium-test-project"
+	userclusterName = "cilium-test-usercluster"
+	ciliumTestNs    = "cilium-test"
+)
+
+func init() {
+	flag.StringVar(&userconfig, "userconfig", "", "path to kubeconfig of usercluster")
+}
+
+func TestReadyCluster(t *testing.T) {
+	if userconfig == "" {
+		t.Logf("kubeconfig for usercluster not provided, test passes vacuously.")
+		t.Logf("to run against ready usercluster use following command")
+		t.Logf("go test ./pkg/test/e2e/cilium -v -race -tags e2e -timeout 30m -run TestReadyCluster -args --userconfig <USERCLUSTER KUBECONFIG>")
+		return
+	}
+	testUserCluster(t, userconfig)
+}
+
+func TestCiliumClusters(t *testing.T) {
+	accessKeyID = os.Getenv("AWS_ACCESS_KEY_ID")
+	if accessKeyID == "" {
+		t.Fatalf("AWS_ACCESS_KEY_ID not set")
+	}
+
+	secretAccessKey = os.Getenv("AWS_SECRET_ACCESS_KEY")
+	if secretAccessKey == "" {
+		t.Fatalf("AWS_SECRET_ACCESS_KEY not set")
+	}
+
+	tests := []struct {
+		name      string
+		proxyMode string
+	}{
+		{
+			name:      "ebpf proxy mode test",
+			proxyMode: resources.EBPFProxyMode,
+		},
+		{
+			name:      "ipvs proxy mode test",
+			proxyMode: resources.IPVSProxyMode,
+		},
+		{
+			name:      "iptables proxy mode test",
+			proxyMode: resources.IPTablesProxyMode,
+		},
+	}
+
+	var mu sync.Mutex
+
+	for _, test := range tests {
+		proxyMode := test.proxyMode
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			mu.Lock()
+			uc, _, cleanup, err := createUsercluster(t, proxyMode)
+			mu.Unlock()
+
+			if err != nil {
+				t.Fatalf("failed to create user cluster: %v", err)
+			}
+
+			defer func() {
+				mu.Lock()
+				cleanup()
+				mu.Unlock()
+			}()
+
+			testUserCluster(t, uc)
+		})
+	}
+}
+
+//gocyclo:ignore
+func testUserCluster(t *testing.T, userconfig string) {
+	config, err := clientcmd.BuildConfigFromFlags("", userconfig)
+	if err != nil {
+		t.Fatalf("failed to build config: %s", err)
+	}
+
+	userClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		t.Fatalf("failed to create client: %s", err)
+	}
+
+	var nodeIP string
+
+	t.Logf("waiting for nodes to come up")
```

Comment:
This I could not use because I have `TestReadyCluster` which can directly test a usercluster if kubeconfig is provided. It would be unnecessarily complex to get projectID, seed etc just to construct the testclient, just to get usercluster client. 

Currently `testUserCluster` only depends on usercluster config which I think is neat.